### PR TITLE
Journalist can create article

### PR DESCRIPTION
**Pull request by mauroavellaneda**
[PT](https://www.pivotaltracker.com/story/show/175137721)
## This PR contains:
- Devise token auth
- User factory bot
- Journalist role
- Create action in articles controller -restricted access to only journalist-
- Content to seeds.rb

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,10 @@
-class ApplicationController < ActionController::Base
+class ApplicationController < ActionController::API
+  include DeviseTokenAuth::Concerns::SetUserByToken
+
+  def error_message(errors)
+    error_message = errors.full_messages.to_sentence
+
+    render json: { message: error_message }, status: 422
+
```
Comment (SebastianN97) [inline app/controllers/application_controller.rb]: Remove one extra space on line 8

**Review by SebastianN97 — APPROVED** [review]

File: db/seeds.rb [hunk 2]
```diff
@@ -1,7 +1,36 @@
...
-# Examples:
-#
-#   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
-#   Character.create(name: 'Luke', movie: movies.first)
+Article.destroy_all
+User.destroy_all
+journalist = User.create(email: "journalist@mail.com", role: "journalist", password: "password")
+registered = User.create(email: "registered@mail.com", role: "registered", password: "password")
+articles = Article.create(
+  [
```
Comment (emtalen) [inline db/seeds.rb:7]: You don't have to wrap them in variables like `articles =` if you're not going to use the variables for anything

File: spec/requests/api/v1/admin/journalist_can_create_articles_spec.rb [hunk 3]
```diff
@@ -0,0 +1,79 @@
+RSpec.describe "POST /v1/admin/articles", type: :request do
+  let(:journalist) { create(:user, role: "journalist") }
+  let(:journalist_credentials) { journalist.create_new_auth_token }
+  let(:journalist_headers) { { HTTP_ACCEPT: "application/json" }.merge!(journalist_credentials) }
+
+  describe "successfully with vaild params and headers" do
+    before do
+      post "/api/v1/admin/articles",
+           params: {
+             title: "Trump holds first public event since Covid diagnosis",
```
Comment (emtalen) [inline spec/requests/api/v1/admin/journalist_can_create_articles_spec.rb]: Please wrap this in a keyword of article, it's cleaner

File: app/controllers/api/v1/admin/articles_controller.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      render json: { message: "Article successfully created" }
+    else
+      error_message(article.errors)
+    end
+  end
+
+  private
+
+  def article_params
+    params.permit(:title, :lead, :content, :category)
```
Comment (emtalen) [inline app/controllers/api/v1/admin/articles_controller.rb]: please send the params in a keyword of article

File: spec/requests/api/v1/admin/journalist_can_create_articles_spec.rb [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    it "article is expected to be associated with journalist" do
+      expect(journalist.articles.first.lead).to eq "Donald Trump has delivered a speech in front of cheering supporters at the White House in his first public appearance since being hospitalised"
+    end
+  end
+
+  describe "unsuccessfully, missing content" do
+    before do
+      post "/api/v1/admin/articles",
+           params: {
+             title: "Trump holds first public event since Covid diagnosis",
```
Comment (emtalen) [inline spec/requests/api/v1/admin/journalist_can_create_articles_spec.rb]: Same here

File: spec/requests/api/v1/admin/journalist_can_create_articles_spec.rb [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+  describe "unauthorized user" do
+    let(:unauthorized_user) { create(:user, role: "registered") }
+    let(:unauthorized_user_credentials) { unauthorized_user.create_new_auth_token }
+    let(:unauthorized_headers) { { HTTP_ACCEPT: "application/json" }.merge!(unauthorized_user_credentials) }
+
+    before do
+      post "/api/v1/admin/articles",
+           params: {
+             title: "Trump holds first public event since Covid diagnosis",
```
Comment (emtalen) [inline spec/requests/api/v1/admin/journalist_can_create_articles_spec.rb]: Same here

**Review by emtalen — CHANGES_REQUESTED** [review]

**Review by emtalen — APPROVED** [review]

**Pull request merged**